### PR TITLE
SB-2 RabbitMQ/DataProcessing

### PR DESCRIPTION
**Pull request by 1jeanpaul1**
## Summary

* RabbitMQ setup for Generate Listings API and message processing.

## Changes
```src/main/java/com/fcgl/endpoints/ManageListing.java```
* Added an endpoint for generating a listing
    * Creates a **ReceiveListingMessage** Object and calls the **processMessages()** method which activates RabbitMQ. Returns a **Response Object**

```src/main/java/com/fcgl/Listing/MessageQueueReceiveListing/ReceiveListingMessages.java```
* This is where messages are received from RabbitMQ. It puts all the messages into a List<String> (Which should be an object representation of the Object now that I think about it). It then calls the **MessageProcessor** class which processes the messages.

```src/main/java/com/fcgl/Listing/MessageQueueReceiveListing/MessageProcessor.java```
* Processes the messages from RabbitMQ. Messages are mapped to IProductInformation Objects
* Note:I'm almost certain there is a better way of mapping Json to objects. I probably have to create a POJO representation and then map it that way. I'll look into that and see what's good.

```src/main/java/com/fcgl/Listing/Vendors/Factories/VendorListingFactory.java```
* Not really a factory in the sense that I am not returning an Object from the same interface. This is basically a task starter. It starts the generate listing task for a particular vendor. 
* This will allow us to have completely different Object representations for Vendor Listing Objects if needed.

## Testing:

Will add instructions on testing manually

### CODE REVIEW

File: src/main/java/com/fcgl/Listing/MessageQueueReceiveListing/ReceiveListingMessages.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+    /**
+     * Receives messages from RabbitMQ and adds then to the messages list
+     * @param messageCount: How many messages will be received
+     * @param channel: The channel getting the messages from RabbitMQ
+     * @param queueName: The name of the queue receiving messages from
+     */
+    private void handleDelivery(int messageCount, Channel channel, String queueName) {
+        for (int i = 0; i < messageCount; i++) {
+            Boolean success = false;
```
Comment (Bigg-T) [inline src/main/java/com/fcgl/Listing/MessageQueueReceiveListing/ReceiveListingMessages.java]: Why Object primitive?

File: src/main/java/com/fcgl/Listing/MessageQueueReceiveListing/Response/MessageProcessorResponse.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+
+public class MessageProcessorResponse {
+    private HashMap<Vendor,ArrayList<IProductInformation>> vendorProductInformation;
+    private List<String> badRequests;
+    private List<Vendor> vendors;
+
+    public MessageProcessorResponse(HashMap<Vendor,ArrayList<IProductInformation>> vendorProductInformation, List<String> badRequests, List<Vendor> vendors) {
```
Comment (Bigg-T) [inline src/main/java/com/fcgl/Listing/MessageQueueReceiveListing/Response/MessageProcessorResponse.java]: too many char in a line.

**Review by Bigg-T — DISMISSED** [review]
Code Style tabs = 2 spaces. Don't use * import. Use the google style style guide as agreed.

File: src/main/java/com/fcgl/Listing/MessageQueueReceiveListing/ReceiveListingMessages.java (hunk 1 above)
Comment (1jeanpaul1) [reply src/main/java/com/fcgl/Listing/MessageQueueReceiveListing/ReceiveListingMessages.java]: No reason

**Comment by 1jeanpaul1** [comment]
@Bigg-T right my bad. Forgot to add the style guide before the PR

File: src/test/java/com/fcgl/Listing/MessageQueueReceiveListing/MessageProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    String message2 = generateJsonFormattedMessage(values2, parameters);
+    messages.add(message1);
+    messages.add(message2);
+    MessageProcessor messageProcessor = new MessageProcessor(messages);
+    MessageProcessorResponse response = messageProcessor.processMessages();
+    assertEquals(1, response.getBadRequests().size());
+    assertEquals(1, response.getVendorProductInformation().get(Vendor.AMAZON).size());
+  }
+
+  private String generateJsonFormattedMessage(String[] message, String[] parameters) {
```
Comment (firstcustomergoodluck) [inline src/test/java/com/fcgl/Listing/MessageQueueReceiveListing/MessageProcessorTest.java:154]: Can probably use gson for this instead. Not a deal breaker. Can do this in another PR

**Comment by firstcustomergoodluck** [comment]
Looks good, made a small comment about tests but seems good to go.

**Review by firstcustomergoodluck — APPROVED** [review]
